chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -1,15 +1,106 @@
 #[cfg(feature = "evm")]
-use chain_evm::state::{AccountTrie, Balance};
+use super::Error;
+#[cfg(feature = "evm")]
+use crate::smartcontract::Contract;
+#[cfg(feature = "evm")]
+use chain_evm::{
+    machine::{Config, Environment, Log, VirtualMachine},
+    state::{AccountTrie, Balance},
+};
 
 #[derive(Default, Clone, PartialEq, Eq)]
 pub struct Ledger {
     #[cfg(feature = "evm")]
     pub(crate) accounts: AccountTrie,
+    #[cfg(feature = "evm")]
+    pub(crate) logs: Vec<Log>,
 }
 
 impl Ledger {
     pub fn new() -> Self {
-        Default::default()
+        Self {
+            #[cfg(feature = "evm")]
+            accounts: Default::default(),
+            #[cfg(feature = "evm")]
+            logs: Default::default(),
+        }
+    }
+    #[cfg(feature = "evm")]
+    pub fn deploy_contract<'runtime>(
+        &mut self,
+        contract: Contract,
+        config: &'runtime Config,
+        environment: &'runtime Environment,
+    ) -> Result<(), Error> {
+        let mut vm = self.virtual_machine(config, environment);
+        match contract {
+            Contract::Create {
+                caller,
+                value,
+                init_code,
+                gas_limit,
+                access_list,
+            } => {
+                //
+                if let Some((new_state, logs)) =
```

Comment:
where do we check the transaction was correctly signed by the caller? or is this left for future revisions?

Reply:
This has not been defined yet.